src/nunit.analyzers/DisposeFieldsInTearDown/DisposeFieldsInTearDownAnalyzer.cs

```diff
@@ -0,0 +1,442 @@
+                {
+                    IMethodSymbol? calledMethod = model.GetSymbolInfo(invocationExpression).Symbol as IMethodSymbol;
+                    if (calledMethod is not null)
+                    {
+                        if (SymbolEqualityComparer.Default.Equals(calledMethod.ContainingType, type))
+                        {
+                            // We are calling a local method on our class, keep looking for assignments.
+                            return DisposedIn(model, type, symbols, calledMethod);
+                        }
+                    }
+                }
+            }
+            else if (expression is ConditionalAccessExpressionSyntax conditionalAccessExpression &&
+                conditionalAccessExpression.WhenNotNull is InvocationExpressionSyntax conditionalInvocationExpression &&
+                conditionalInvocationExpression.Expression is MemberBindingExpressionSyntax memberBindingExpression &&
+                IsDispose(memberBindingExpression.Name))
+            {
+                string? target = GetTargetName(conditionalAccessExpression.Expression);
+                if (target is not null && symbols.Contains(target))
+                {
+                    disposedSymbols.Add(target);
+                }
+            }
+
+            return disposedSymbols;
+        }
+
+        private static HashSet<string> DisposedIn(SemanticModel model, INamedTypeSymbol type, HashSet<string> symbols, StatementSyntax statement)
+        {
+            switch (statement)
+            {
+                case ExpressionStatementSyntax expressionStatement:
+                    return DisposedIn(model, type, symbols, expressionStatement.Expression);
+
+                case IfStatementSyntax ifStatement:
+                    {
+                        // We don't care about the condition
+                        HashSet<string> disposedSymbolsInStatement = DisposedIn(model, type, symbols, ifStatement.Statement);
+                        if (ifStatement.Else is not null)
+                            disposedSymbolsInStatement.UnionWith(DisposedIn(model, type, symbols, ifStatement.Else.Statement));
+
+                        return disposedSymbolsInStatement;
+                    }
+
+                case BlockSyntax block:
+                    return DisposedIn(model, type, symbols, block.Statements);
+
+                case SwitchStatementSyntax switchStatement:
+                    {
+                        var disposedSymbols = new HashSet<string>();
+
+                        foreach (var caseStatements in switchStatement.Sections.Select(x => x.Statements))
+                        {
+                            HashSet<string> disposedSymbolsInStatement = DisposedIn(model, type, symbols, caseStatements);
+                            disposedSymbols.UnionWith(disposedSymbolsInStatement);
+                        }
+
+                        return disposedSymbols;
+                    }
```

Comment:
```suggestion
                case SwitchStatementSyntax switchStatement:
                {
                    var disposedSymbols = new HashSet<string>();

                    foreach (var caseStatements in switchStatement.Sections.Select(x => x.Statements))
                    {
                        HashSet<string> disposedSymbolsInStatement = DisposedIn(model, type, symbols, caseStatements);
                        disposedSymbols.UnionWith(disposedSymbolsInStatement);
                    }

                    return disposedSymbols;
                }
```

Reply:
This was autoformatted when I copied this.
I have add `csharp_indent_case_contents_when_block = false` to our `.editorconfig` to match the wanted style.
It didn't affect any other code as there were no other cases with curly braces.